### PR TITLE
Handle errors on EncryptedMailAddress

### LINKED ISSUE #2649: Cannot open broken invites
Users report this for accepted invites regularly, seems like mail address of one of the recipients is invalid. We should handle it.

```
Tutanota version: 3.80.4
TypeError
Error message: Cannot read property 'endsWith' of undefined
Stacktrace:
TypeError: Cannot read property 'endsWith' of undefined
at file:///Applications/Tutanota%20Desktop.app/Contents/Resources/app.asar/common.js:1:12959
at file:///Applications/Tutanota%20Desktop.app/Contents/Resources/app.asar/main.js:1:2013187
at SendMailModel.value (file:///Applications/Tutanota%20Desktop.app/Contents/Resources/app.asar/main.js:1:1329433)
at SendMailModel.value (file:///Applications/Tutanota%20Desktop.app/Contents/Resources/app.asar/main.js:1:1328987)
at file:///Applications/Tutanota%20Desktop.app/Contents/Resources/app.asar/main.js:1:863447
at Array.forEach ()
at CalendarEventViewModel.value (file:///Applications/Tutanota%20Desktop.app/Contents/Resources/app.asar/main.js:1:863285)
at new CalendarEventViewModel (file:///Applications/Tutanota%20Desktop.app/Contents/Resources/app.asar/main.js:1:859975)
at file:///Applications/Tutanota%20Desktop.app/Contents/Resources/app.asar/main.js:1:853688
at new CalendarEventPopup (file:///Applications/Tutanota%2
```

> I received an event update by email that I accepted. Since then, I cannot view the event in question in my calendar anymore. Each time I try, I get the error message which led me to send you this report.

> Cannot open this agenda item

> when accepting a google calendar event and opening the event to edit, the error started to appear and now I can't edit it nor delete it.

> Vom Absender gelöschte Termine verschwinden nicht vom Kalender.

> I am simply clicking on an accepted event on my calendar but I cannot access the event or make any changes. The orignal sender is on Google. THis same error occurs on both Linux and Windows desktop apps and the iOS app.

> Hello, some items in my calendar all of a sudden cannot be open, it notifies me of an error. It´s very uncomfortable as I have links to meetings in some of those items. Thank you. And pls consider making a bigger place for this message, not only one line.

### PR DESCRIPTION
**Pull request by bedhub**
handle crypt errors on calendar event attendees  fixes #2649

### CODE REVIEW

**Review by charlag — APPROVED** [review]

**Comment by vaf-hub** [comment]
I think we should also handle this case in the `CalendarEventPopup` which would just show the status icon in case the mail address is set to null. And we should consider indicating to the user that there was an attendee that could not be parsed correctly instead of just not showing them.

File: src/calendar/view/EventPreviewView.js [hunk 1]
```diff
@@ -78,13 +79,17 @@ export class EventPreviewView implements MComponent<Attrs> {
 	}
 
 	_renderAttendee(attendee: CalendarEventAttendee): Children {
+		let attendeeField = attendee.address.address
+		if(downcast(attendee.address)._errors){
```
Comment (charlag) [inline src/calendar/view/EventPreviewView.js:83]: We should make some utility function to check for errors.

Comment (vaf-hub) [reply src/calendar/view/EventPreviewView.js:83]: Do you mean just for the EncryptedMailAddressType or more generic?

Comment (charlag) [reply src/calendar/view/EventPreviewView.js:83]: For any entity, yes

Comment (vaf-hub) [reply src/calendar/view/EventPreviewView.js:83]: Like this?
`export function hasError<T, K>(instance: T, key: ?K): boolean {
	return !instance || (!!instance._errors && (!key || !!instance._errors.key))
}`

Comment (charlag) [reply src/calendar/view/EventPreviewView.js:83]: Yeah except if you don't restrict `T` it's effectively `any`. Maybe you could use `SomeEntity` if it works.

Comment (vaf-hub) [reply src/calendar/view/EventPreviewView.js]: Yeah then we need `downcast()`. I pushed it.

**Pull request merged**